broker-core/src/main/java/io/zeebe/broker/workflow/processor/handlers/element/ElementTerminatedHandler.java

```diff
@@ -49,6 +53,7 @@ protected boolean handleState(BpmnStepContext<T> context) {
             && WorkflowInstanceLifecycle.canTransition(
```

Comment:
I know, it was a horrible mistake on my part as they're also impossible to test :disappointed_relieved:

Reply:
So is it OK for you if we try to refactor this a bit in Q3? So use composition instead?

Reply:
I would love to, I already mentioned it to everyone who wants to hear about it :sweat_smile: But unfortunately it's not in our OKRs, so it would have to be something we do on slack days or something.

Reply:
Yeah that's why I meant Q3.  :sweat_smile: I assume that we will not change much in the processors next quarter

Reply:
Oh, my bad, yes I hope! Although I don't know if I can't wait that long, it bothers me too much :sweat_smile: esp. since it's my fault

Reply:
hehe, no problem in this way we learn :slightly_smiling_face: